audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/request/ReadRequestCompleteAuditRequest.java

```diff
@@ -0,0 +1,160 @@
+        this.rulesApplied = requireNonNull(rulesApplied);
+        this.numberOfRecordsReturned = numberOfRecordsReturned;
+        this.numberOfRecordsProcessed = numberOfRecordsProcessed;
+    }
+
+    public interface IUser {
+        /**
+         * @param user {@link User} is the user that made the initial registration request to access data
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        ILeafResource withUser(final User user);
+    }
+
+    public interface ILeafResource {
+        /**
+         * @param leafResource the {@link LeafResource} which the data has just finished being read
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        IContext withLeafResource(final LeafResource leafResource);
+    }
+
+    public interface IContext {
+        /**
+         * @param context the context that was passed by the client to the palisade service
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        IRulesApplied withContext(final Context context);
+    }
+
+    public interface IRulesApplied {
+        /**
+         * @param rules {@link Rules} is the rules that are being applied to this resource for this request
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        INumberOfRecordsReturned withRulesApplied(final Rules rules);
+    }
+
+    public interface INumberOfRecordsReturned {
+        /**
+         * @param numberOfRecordsReturned is the number of records that was returned to the user from this resource
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        INumberOfRecordsProcessed withNumberOfRecordsReturned(final long numberOfRecordsReturned);
+    }
+
+    public interface INumberOfRecordsProcessed {
+        /**
+         * @param numberOfRecordsProcessed is the number of records that was processed from this resource
+         * @return the {@link ReadRequestCompleteAuditRequest}
+         */
+        ReadRequestCompleteAuditRequest withNumberOfRecordsProcessed(final long numberOfRecordsProcessed);
+    }
+
+    /**
+     * Static factory method.
+     *
+     * @param original the original request id
+     * @return {@link ReadRequestCompleteAuditRequest}
+     */
+    public static IUser create(final RequestId original) {
+        return user -> leafResource -> context -> rulesApplied -> numberOfRecordsReturned -> numberOfRecordsProcessed -> new ReadRequestCompleteAuditRequest(null, original, user, leafResource, context, rulesApplied, numberOfRecordsReturned, numberOfRecordsProcessed);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final ReadRequestCompleteAuditRequest that = (ReadRequestCompleteAuditRequest) o;
+        return new EqualsBuilder()
```

Comment:
Remove dependencies on EqualsBuilder() and use the intellij tool to generate the code